src/biogeochem/CNVegNitrogenStateType.F90

```diff
@@ -2276,6 +2304,77 @@ subroutine Summary_nitrogenstate(this, bounds, num_soilc, filter_soilc, num_soil
 
   end subroutine Summary_nitrogenstate
 
+  !-----------------------------------------------------------------------
+  subroutine ReadParams ( this, ncid )
+    !
+    ! !USES:
+    use paramUtilMod, only: readNcdioScalar
+
+    implicit none
+
+    ! !ARGUMENTS:
+    class(cnveg_nitrogenstate_type) :: this
+    type(file_desc_t), intent(inout) :: ncid  ! pio netCDF file id
+    !
+    ! !LOCAL VARIABLES:
+    !-----------------------------------------------------------------------
+
+    ! read in parameters
+
+    ! read in the scalar parameters
+
+    ! ppm CO2 for target leaf CN scaling (ppmv)
+    call readNcdioScalar(ncid, 'leafcn_co2_base', sourcefile, params_inst%leafcn_co2_base)
+    ! CO2 effects on target leaf CN scaling (unitless)
+    call readNcdioScalar(ncid, 'leafcn_co2_slope', sourcefile, params_inst%leafcn_co2_slope)
+
+  end subroutine ReadParams
+
+  !-----------------------------------------------------------------------
+  subroutine time_evolv_leafcn(this, bounds, atm2lnd_inst)
+    !
+    ! AUTHOR: slevis
+    !
+    ! !DESCRIPTION:
+    ! Update time-evolving parameters
+    ! In the current implementation, I call this subroutine from clm_drv
+    ! first thing, before anything else. Hence I allocate leafcn_patch
+    ! to memory if_first_step here to prevent an error in some tests.
+    ! I allocate again in InitAllocate to prevent a similar error in
+    ! InitHistory in other tests.
+    !
+    ! !USES:
+    use pftconMod, only : pftcon
+    use clm_time_manager, only : is_first_step
+    !
+    ! !ARGUMENTS:
+    class(cnveg_nitrogenstate_type) :: this
+    type(bounds_type) , intent(in)  :: bounds
+    type(atm2lnd_type), intent(in)  :: atm2lnd_inst
+    !
+    ! !LOCAL VARIABLES:
+    integer :: g, l, c, p  ! indices
+    real(r8) :: co2_ppmv  ! atm co2 concentration
+    !-----------------------------------------------------------------------
+
+    if (is_first_step()) then
+       allocate(this%leafcn_patch(bounds%begp:bounds%endp)); this%leafcn_patch(:) = pftcon%leafcn(patch%itype(:))
```

Comment:
Replace the allocate with an endrun as a test to confirm whether I really need to allocate here, because I shouldn't have to.